### PR TITLE
feat: Navigation from the Editor to Snyk ToolWindow [ROAD-887]

### PR DESCRIPTION
**Pull request by ArtsiomCh**
feat: Navigation from the Editor (using QuickFix's functionality atnnotations) to Snyk ToolWindow (correspondent node in the Tree opened and Description shown) for:

  - Open Source scan results;
  - SnykCode scan results;
  - Infrastructure as a Code scan results;
  - Container scan results.

fix: Some minor UI renaming and alignments
image
image

### CODE REVIEW

File: src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanelIntegTest.kt [hunk 1]
```diff
@@ -772,4 +818,146 @@ class SnykToolWindowPanelIntegTest : HeavyPlatformTestCase() {
...
+        val firstOssFileNode = rootOssIssuesTreeNode.firstChild as FileTreeNode
+        val firstOssIssueNode = firstOssFileNode.firstChild as VulnerabilityTreeNode
+        val groupedVulns = firstOssIssueNode.userObject as Collection<Vulnerability>
+        val vulnerability = groupedVulns.first()
+
+        // actual test run
+        toolWindowPanel.selectNodeAndDisplayDescription(vulnerability)
+        // hack to avoid "File accessed outside allowed roots" check in tests
+        // needed due to com.intellij.openapi.vfs.newvfs.impl.VfsRootAccess.assertAccessInTests
+        val prev_isInStressTest = ApplicationInfoImpl.isInStressTest()
```
Comment (bastiandoetsch) [inline src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanelIntegTest.kt:873]: same here - also, can we maybe de-duplicate this?

File: src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanelIntegTest.kt [hunk 2]
```diff
@@ -772,4 +818,146 @@ class SnykToolWindowPanelIntegTest : HeavyPlatformTestCase() {
...
+        PlatformTestUtil.dispatchAllEventsInIdeEventQueue()
+        val rootContainerTreeNode = toolWindowPanel.getRootContainerIssuesTreeNode()
+        val firstImageNode = rootContainerTreeNode.firstChild as ContainerImageTreeNode
+        val containerImage = firstImageNode.userObject as ContainerIssuesForImage
+
+        // actual test run
+        toolWindowPanel.selectNodeAndDisplayDescription(containerImage)
+        // hack to avoid "File accessed outside allowed roots" check in tests
+        // needed due to com.intellij.openapi.vfs.newvfs.impl.VfsRootAccess.assertAccessInTests
+        val prev_isInStressTest = ApplicationInfoImpl.isInStressTest()
```
Comment (bastiandoetsch) [inline src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanelIntegTest.kt:944]: naming & deduplication :)

File: src/main/kotlin/snyk/common/intentionactions/AlwaysAvailableReplacementIntentionAction.kt [hunk 3]
```diff
@@ -65,7 +48,7 @@ class AlwaysAvailableReplacementIntentionAction(
     }
 
     companion object {
-        private const val intentionDefaultTextPrefix = "Change to "
-        private const val intentionDefaultFamilyName = "Snyk"
+        private const val intentionTextPrefix = "Upgrade to "
+        private const val intentionTextPostfix = " (Snyk)"
```
Comment (bastiandoetsch) [inline src/main/kotlin/snyk/common/intentionactions/AlwaysAvailableReplacementIntentionAction.kt:52]: Hm... I find it easier to have `Snyk` as a prefix, but I suppose that's from UX?

File: src/main/kotlin/snyk/common/intentionactions/SnykIntentionActionBase.kt [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Iconable
+import com.intellij.psi.PsiFile
+
+abstract class SnykIntentionActionBase() : IntentionAction, Iconable, PriorityAction {
+
+    override fun startInWriteAction(): Boolean = true
+
+    override fun getFamilyName(): String  = intentionDefaultFamilyName
```
Comment (bastiandoetsch) [inline src/main/kotlin/snyk/common/intentionactions/SnykIntentionActionBase.kt:14]: Pls format the file :)

File: src/main/kotlin/snyk/oss/annotator/OSSBaseAnnotator.kt [hunk 5]
```diff
@@ -27,26 +30,32 @@ abstract class OSSBaseAnnotator : ExternalAnnotator<PsiFile, Unit>() {
...
+            .distinctBy { getIntroducingPackage(it) + it.id }
+
+        filteredVulns.forEach { vulnerability ->
+            if (vulnerability.ignored || vulnerability.obsolete) return@forEach
+            val textRange = textRange(psiFile, vulnerability)
+            val highlightSeverity = vulnerability.getSeverity().getHighlightSeverity()
+            if (textRange != TextRange.EMPTY_RANGE) {
+                val annotationMessage = annotationMessage(vulnerability)
+                val annotationBuilder =
+                    holder.newAnnotation(highlightSeverity, "Snyk: $annotationMessage").range(textRange)
```
Comment (bastiandoetsch) [inline src/main/kotlin/snyk/oss/annotator/OSSBaseAnnotator.kt:44]: Why is `Snyk` a prefix here and not a postfix `(Snyk)`

**Review by bastiandoetsch — APPROVED** [review]
Love it! Minor suggestions/changes, but nothing wild.

File: src/main/kotlin/snyk/common/intentionactions/AlwaysAvailableReplacementIntentionAction.kt (hunk 3 above)
Comment (ArtsiomCh) [reply src/main/kotlin/snyk/common/intentionactions/AlwaysAvailableReplacementIntentionAction.kt:52]: Yep, it's UX. Andy suggested to place it like that for readability/clarity

File: src/main/kotlin/snyk/iac/annotator/IacBaseAnnotator.kt [hunk 6]
```diff
@@ -43,8 +46,10 @@ abstract class IacBaseAnnotator : ExternalAnnotator<PsiFile, Unit>() {
             .forEach { iacIssue ->
                 LOG.debug("-> ${iacIssue.id}: ${iacIssue.title}: ${iacIssue.lineNumber}")
                 val highlightSeverity = iacIssue.getSeverity().getHighlightSeverity()
-                holder.newAnnotation(highlightSeverity, annotationMessage(iacIssue))
+                val annotationMessage = annotationMessage(iacIssue)
+                holder.newAnnotation(highlightSeverity, "Snyk: $annotationMessage")
```
[the remark this answers is not in the document]
Comment (ArtsiomCh) [reply src/main/kotlin/snyk/iac/annotator/IacBaseAnnotator.kt:50]: UX by Andy again :) 
Here we could have numerous annotations form Snyk and not. In Quick fixes it's less likely and in fixes user should be focused on the action and not where the proposal come from.
PS that's how i understood it.

[the remark this answers is not in the document]
Comment (bastiandoetsch) [reply src/main/kotlin/snyk/iac/annotator/IacBaseAnnotator.kt:50]: @andygongea This UX is a bit inconsistent - could you please have a look? I think we should either use a pre- or a postfix for all quickfixes/annotations.

**Pull request merged**